experimental/single_number_formatter/src/ule.rs

```diff
@@ -0,0 +1,112 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use zerovec::ule::{AsULE, ZeroVecError, ULE};
+
+use crate::provider::{CurrencyPatterns, PatternSelection};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[repr(transparent)]
+pub struct CurrencyPatternsULE([u8; 3]);
+
+impl CurrencyPatternsULE {}
+
+// Safety (based on the safety checklist on the ULE trait):
+//  1. CurrencyPatternsULE does not include any uninitialized or padding bytes.
+//     (achieved by `#[repr(transparent)]` on a ULE type)
+//  2. CurrencyPatternsULE is aligned to 1 byte.
+//     (achieved by `#[repr(transparent)]` on a ULE type)
+//  3. The impl of validate_byte_slice() returns an error if any byte is not valid.
+//  4. The impl of validate_byte_slice() returns an error if there are extra bytes.
+//  5. The other ULE methods use the default impl.
+//  6. CurrencyPatternsULE byte equality is semantic equality.
+unsafe impl ULE for CurrencyPatternsULE {
+    fn validate_byte_slice(bytes: &[u8]) -> Result<(), ZeroVecError> {
+        if bytes.len() % 3 != 0 {
+            return Err(ZeroVecError::length::<Self>(bytes.len()));
+        }
+
+        // #[allow(clippy::indexing_slicing)] // chunks
+        // if !bytes
+        //     .chunks(3)
+        //     .all(|c| Self::bytes_in_range((&c[0], &c[1], &c[2])))
+        // {
+        //     return Err(ZeroVecError::parse::<Self>());
+        // }
+        Ok(())
+    }
+}
+
+impl AsULE for CurrencyPatterns {
+    type ULE = CurrencyPatternsULE;
+
+    #[inline]
+    fn to_unaligned(self) -> Self::ULE {
+        let mut first_byte: u8 = 0;
+
+        if self.short_pattern_standard == PatternSelection::StandardAlphaNextToNumber {
+            first_byte |= 0b0100_0000;
+        }
+        if self.narrow_pattern_standard == PatternSelection::StandardAlphaNextToNumber {
+            first_byte |= 0b0001_0000;
+        }
```

Comment:
Thought: Did you choose these values to reserve two bits for adding more pattern variants, in case we gain a 3rd or 4th pattern variant?